### PR TITLE
basic anchor tests & start get underlying price

### PR DESCRIPTION
**Pull request by maxwolff**
* mocked getAnchorPrice while it was being finalized
* reproduced the basic anchor tests from before, starting to test getUnderlyingPrice and the how we handle priceSources

### CODE REVIEW

File: tests/UniswapAnchoredViewTest.js [hunk 1]
```diff
@@ -0,0 +1,158 @@
+const { encode, sign } = require('../sdk/javascript/.tsbuilt/reporter');
+const { uint, keccak256, time, numToHex, address } = require('./Helpers');
+
+async function setupTokenPairs() {
```
Comment (antonina-cherednichenko) [inline tests/UniswapAnchoredViewTest.js:4]: Maybe, let's delete these token pairs or simplify their data from here and just use mockedVersion. And then real-world data will be in a second file. 

Or can clean up this thing later.

File: contracts/Uniswap/UniswapAnchoredView.sol [hunk 2]
```diff
@@ -120,7 +120,7 @@ contract UniswapAnchoredView is UniswapConfig {
      */
     function getUnderlyingPrice(address cToken) public view returns (uint) {
         TokenConfig memory config = getTokenConfigByCToken(cToken);
-        return priceInternal(config);
+        return mul(1e30, priceInternal(config)) / config.baseUnit;
```
Comment (jflatow) [inline contracts/Uniswap/UniswapAnchoredView.sol:123]: 🎣

File: tests/Helpers.js [hunk 3]
```diff
@@ -58,5 +65,7 @@ module.exports = {
 	time,
 	numToBigNum,
 	numToHex,
-	uint256
+	uint256,
+  uint,
+  keccak256
```
Comment (jflatow) [inline tests/Helpers.js:70]: is this just missing indentation?

File: tests/UniswapAnchoredViewTest.js [hunk 4]
```diff
@@ -0,0 +1,172 @@
...
+      await send(uniswapAnchoredView, 'setAnchorPrice', ['ETH', 110.1e6]);
+      const tx = await postPrices(timestamp, [[['ETH', 100]]], ['ETH']);
+
+      expect(tx.events.PriceGuarded).not.toBe(undefined);
+      expect(tx.events.PriceUpdated).toBe(undefined);
+      expect(await call(uniswapAnchoredView, 'prices', [keccak256('ETH')])).numEquals(0);
+      expect(await call(priceData, 'getPrice', [reporter.address, 'ETH'])).numEquals(100e6);
+    });
+
+    it.todo('test anchor with non-eth prices')
```
Comment (jflatow) [inline tests/UniswapAnchoredViewTest.js:117]: This isn't valid I don't think

File: tests/Helpers.js (hunk 3 above)
Comment (maxwolff) [reply tests/Helpers.js:70]: yes, oops

File: tests/UniswapAnchoredViewTest.js (hunk 4 above)
Comment (maxwolff) [reply tests/UniswapAnchoredViewTest.js:117]: hm?

File: tests/Helpers.js [hunk 5]
```diff
@@ -58,5 +65,7 @@ module.exports = {
 	time,
 	numToBigNum,
 	numToHex,
-	uint256
+	uint256,
+  uint,
```
Comment (maxwolff) [inline tests/Helpers.js:69]:
```suggestion
     uint,
```

File: tests/UniswapAnchoredViewTest.js (hunk 4 above)
Comment (jflatow) [reply tests/UniswapAnchoredViewTest.js:117]: One half the Uniswap pair *must* be eth for the spec / code

Comment (maxwolff) [reply tests/UniswapAnchoredViewTest.js:117]: ha, yes just meant REP-ETH instead of ETH-USD

**Pull request merged**